pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -705,37 +701,64 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 					if err != nil {
 						return err
 					}
+					// In case Portworx if the snapshot ID is populated for every volume then the snapshot
+					// process is considered to be completed successfully.
+					// This ensures we don't execute the post-exec before all volume's snapshot is completed
+					if driverName == volume.PortworxDriverName {
+						for {
+							// Get fresh stock of the status for all volumes.
+							snapshotNotCompleted := false
+							volInfos, err := driver.GetBackupStatus(backup)
+							if err != nil {
+								return fmt.Errorf("error getting backup status for driver %v: %v", driverName, err)
+							}
+							for _, volInfo := range volInfos {
+								if volInfo.BackupID == "" {
+									log.ApplicationBackupLog(backup).Tracef("Snapshot of volume %v is not done yet, need to loop till snapshot is done...", volInfo.PersistentVolumeClaim)
+									snapshotNotCompleted = true
+								}
+							}
```

Comment:
Can we add a logic to add the CR update to avoid px-backup 30 mint timeout as it is a infinite for loop?